lib/spack/spack/build_systems/r.py

```diff
@@ -33,6 +41,34 @@ class RPackage(PackageBase):
 
     extends('r')
 
+    @property
+    def homepage(self):
+        if self.cran:
+            return 'https://cloud.r-project.org/package=' + self.cran
+        elif self.bioc:
+            return 'https://bioconductor.org/packages/' + self.bioc
+
+    @property
+    def url(self):
+        if self.cran:
+            return (
+                'https://cloud.r-project.org/src/contrib/'
+                + self.cran + '_' + str(list(self.versions)[0]) + '.tar.gz'
```

Comment:
Does this work during `spack create`? I'm not sure if `self.url` is accessed before we officially have a list of versions.

Reply:
I tested with a CRAN package that is not in spack already using the following commands:

- `spack create https://cran.r-project.org/src/contrib/BinGSD_0.0.1.tar.gz`
- `spack info r-bingsd`
- `spack stage r-bingsd`

All of those worked fine. I am not sure how to verify for `spack url summary`.

Reply:
The `BinGSD` example that I grabbed is a new CRAN package and does not have an Archive directory yet. So, I also modified `r-caret` to use the `cran` attribute and tested 

- `spack versions r-caret`
- `spack checksum r-caret <<<2`

Those worked fine.

Reply:
> I am not sure how to verify for `spack url summary`

Basically you can run `spack url summary` on develop and on your branch and they should be similar/the same for version detection.

Reply:
I get the same exact output for both branches.